lib/heidrun/id_source.rb

```diff
@@ -0,0 +1,55 @@
+
+module Heidrun
+
+  ##
+  # A source of identifier strings to use in constructing API requests.
+  #
+  # This is a temporary trick to allow us to get around a limitation
+  # in NARA's API: we need to request a list of IDs explicitly because we can't
+  # page through their entire result set.  IDSource reads a file and gives us
+  # batches of IDs as arrays.
+  #
+  class IDSource
+
+    ##
+    # Constructor
+    #
+    # IDSource is initialized with the filehandle of a file that contains one
+    # ID string per line.
+    #
+    # The batchsize parameter will determine how many records are requested
+    # with a single request to the API.
+    #
+    # @param fh        File    File-like object (File / StringIO)
```

Comment:
Reformat type specs here and on `#batches` to include square brackets.

Also, should the type spec here be `[IO]` or maybe `[#each_line]`?